framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java

```diff
@@ -629,7 +629,7 @@ public static AnnotatedTypeMirror leastUpperBound(
 
         parameters = new ArrayList<>(parameters.subList(0, parameters.size() - 1));
         for (int i = args.size() - parameters.size(); i > 0; --i) {
-            parameters.add(varargs.getComponentType());
+            parameters.add(varargs.getComponentType().shallowCopy());
```

Comment:
Why is a shallow copy enough?
What happens with a poly annotation that is on a type argument?
Can you add a test case to ensure this doesn't cause a problem?

Reply:
I expanded the test and verified that shallowCopy doesn't do the right thing. I changed this to deepCopy.